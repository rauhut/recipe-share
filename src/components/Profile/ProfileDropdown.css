.dropdown-menu-right {
  right: 0 !important;
}

.dropdown-menu.show {
  display: flex !important;
  justify-content: center;
}
