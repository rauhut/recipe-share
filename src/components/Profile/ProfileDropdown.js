import React from "react";
import {
  Dropdown,
  DropdownToggle,
  DropdownMenu,
  DropdownItem,
} from "reactstrap";
import "./ProfileDropdown.css";

class ProfileDropdown extends React.Component {
  constructor(props) {
    super(props);
    this.toggle = this.toggle.bind(this);
    this.state = {
      dropdownOpen: false,
    };
  }

  toggle = () => {
    this.setState((prevState) => ({
      dropdownOpen: !prevState.dropdownOpen,
    }));
  };

  render() {
    return (
      <div>
        <Dropdown isOpen={this.state.dropdownOpen} toggle={this.toggle}>
          <DropdownToggle
            tag="span"
            data-toggle="dropdown"
            aria-expanded={this.dropdownOpen}
          >
            <h3>{this.props.user.name}</h3>
          </DropdownToggle>
          <DropdownMenu
            right
            style={{
              backgroundColor: "rgba(255, 255, 255, 0.5)",
            }}
          >
            <DropdownItem>
              <button
                className="button-secondary"
                onClick={() => {
                  const token = window.sessionStorage.getItem("token");
                  fetch("http://localhost:3000/signout", {
                    method: "post",
                    headers: {
                      "Content-Type": "application/json",
                      Authorization: token,
                    },
                  })
                    .then((response) => {
                      if (response.ok) {
                        return response;
                      } else {
                        throw new Error(
                          "Something went wrong when trying to logout"
                        );
                      }
                    })
                    .catch(console.log);
                  window.sessionStorage.removeItem("token");
                  this.props.onRouteChange("signout");
                }}
              >
                Sign Out
              </button>
            </DropdownItem>
          </DropdownMenu>
        </Dropdown>
      </div>
    );
  }
}

export default ProfileDropdown;
