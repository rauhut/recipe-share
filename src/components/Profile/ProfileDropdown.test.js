import React from "react";
import { shallow } from "enzyme";
import ProfileDropdown from "./ProfileDropdown";

describe("ProfileDropdown", () => {
  let wrapper;

  const mockProps = {
    user: {
      id: "123456789",
      username: "testuser",
      recipes: "",
    },
  };

  it("renders for user", () => {
    wrapper = shallow(<ProfileDropdown {...mockProps} />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
