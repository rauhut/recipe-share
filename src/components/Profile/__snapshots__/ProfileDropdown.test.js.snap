// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileDropdown renders for user 1`] = `
"<div>
  <Dropdown isOpen={false} toggle={[Function: bound ]} a11y={true} direction=\\"down\\" nav={false} active={false} addonType={false} inNavbar={false} setActiveFromChild={false}>
    <DropdownToggle tag=\\"span\\" data-toggle=\\"dropdown\\" aria-expanded={[undefined]} aria-haspopup={true} color=\\"secondary\\">
      <h3>
        testuser
      </h3>
    </DropdownToggle>
    <DropdownMenu right={true} style={{...}} tag=\\"div\\" flip={true}>
      <DropdownItem tag=\\"button\\" toggle={true}>
        <button className=\\"button-secondary\\" onClick={[Function: onClick]}>
          Sign Out
        </button>
      </DropdownItem>
    </DropdownMenu>
  </Dropdown>
</div>"
`;
